feat(integrations): add Agno integration docs (#1529)
apify/apify-docs: sources/platform/integrations/ai/agno.md

---

> title: Agno Integration

Reviewer: Please add Agno [here](https://github.com/apify/apify-docs/blob/master/.github/styles/config/vocabularies/Docs/accept.txt) to add it to Vale's vocabulary list

sidebar_label: Agno
description: Integrate Apify with Agno to power AI agents with web scraping, automation, and data insights.
sidebar_position: 1
slug: /integrations/agno
---

**Integrate Apify with Agno to power AI agents with web scraping, automation, and data insights.**

---

## What is Agno?

[Agno](https://docs.agno.com/) is an open-source framework for building intelligent AI agents. It provides a flexible architecture to create agents with custom tools, enabling seamless integration with external services like Apify for tasks such as web scraping, data extraction and automation.

> [!NOTE]

> > Check out the [Agno documentation](https://docs.agno.com/introduction) for more details on building AI agents.

Reviewer: Please use docusaurus [admonition](https://docusaurus.io/docs/markdown-features/admonitions) as is in our style guide in [CONTRIBUTING](https://github.com/apify/apify-docs/blob/master/CONTRIBUTING.md#formatting-conventions)

## How to Use Apify with Agno

> This guide shows how to integrate Apify Actors with Agno to empower your AI agents with real-time web data. We'll use the [RAG Web Browser](https://apify.com/apify/rag-web-browser) Actor to fetch web content and the [Google Places Crawler](https://apify.com/compass/crawler-google-places) Actor to extract location-based data. It is very easy to use with any other Actor by just passing the name of the Actor. See and choose from 4,000+ Actors in the [Apify Store](https://apify.com/store).

Reviewer: Let's try to avoid using specific numbers, it's a maintenance nightmare otherwise

```suggestion
This guide shows how to integrate Apify Actors with Agno to empower your AI agents with real-time web data. We'll use the [RAG Web Browser](https://apify.com/apify/rag-web-browser) Actor to fetch web content and the [Google Places Crawler](https://apify.com/compass/crawler-google-places) Actor to extract location-based data. It is very easy to use with any other Actor by just passing the name of the Actor. See and choose from thousands of Actors in the [Apify Store](https://apify.com/store).
```

### Prerequisites

- **Apify API token**: Obtain your token from the [Apify console](https://console.apify.com/account/integrations).  
- **Python environment**: Ensure Python is installed (version 3.8+ recommended).  

> - **Required packages**: Install the following dependencies in your terminal:

Reviewer: Please change to emphasis from bold. Bold should be use to highlight UI elements.

```bash

> pip install agno apify-client langchain-apify

Reviewer 2: Why is langchain-apify needed?

Author: It is necessary in the implementation because it enables the agent to use the Actor by providing the actor tool’s name and a brief description. It abstracts the complexity of interacting with Apify Actors, allowing to invoke them directly.

# Create a travel planning agent
agent = Agent(
    name="Travel Planner",
    instructions=[
        "You are a travel planning assistant. Use web data and location insights to provide detailed travel recommendations."
    ],
    tools=[
        ApifyTools(
            actors=[
                "apify/rag-web-browser",           # For general web research
                "compass/crawler-google-places"    # For location-based data
            ],
            apify_api_token="YOUR_APIFY_API_TOKEN"
        )
    ],
    show_tool_calls=True,
    markdown=True
)

# Plan a trip to Tokyo
agent.print_response(
    """
    I'm traveling to Tokyo next month.
    1. Research the best time to visit and top attractions.
    2. Find a highly rated sushi restaurant near Shinjuku.
    Compile a travel guide with this information.
    """,
    markdown=True
)
```

This agent will fetch travel-related data and restaurant recommendations, providing a comprehensive travel guide:

1. Use the RAG Web Browser to research Tokyo travel details.
2. Use the Google Places Crawler to find a top sushi restaurant.
3. Combine the results into a comprehensive guide.

> [!TIP]

> > Browse the [Apify Store](https://apify.com/store) to find additional Actors for tasks like social media scraping, e-commerce data extraction, or news aggregation.

Reviewer: Please use docusaurus [admonition](https://docusaurus.io/docs/markdown-features/admonitions) as is in our style guide in [CONTRIBUTING](https://github.com/apify/apify-docs/blob/master/CONTRIBUTING.md#formatting-conventions)

### Available Apify Tools

Agno supports any Apify Actor via the ApifyTools class. You can specify a single Actor ID or a list of Actor IDs to register multiple tools for your agent at once.

## Configuration Options

| Parameter                    | Type                | Default | Description                                                        |
| ---------------------------- | ------------------- | ------- | ------------------------------------------------------------------ |
| `apify_api_token`            | `str`               | `None`  | Apify API token (or set via APIFY_API_TOKEN environment variable)  |

> | `actors`                     | `str` or `List[str]`| `None`  | Single Actor ID or list of Actor IDs to register                   |

Reviewer 2: The current `ApifyTools` interface looks different. 
[See here](https://github.com/agno-agi/agno/blob/main/libs/agno/agno/tools/apify.py)

Author: That is because we updated both the docs and the actual implementation simultaneously. It's connected with to [PR#2566]https://github.com/agno-agi/agno/pull/2566 which actually has this implementation.

---
title: Agno Integration
sidebar_label: Agno
description: Integrate Apify with Agno to power AI agents with web scraping, automation, and data insights.

> sidebar_position: 1

Reviewer: In a separate [PR](https://github.com/apify/apify-docs/pull/1553) I've cleaned up AI integration sidebar, would you mind fitting it in? We would like to keep MCP as first from the top

sidebar_position: 14
slug: /integrations/agno
---

**Integrate Apify with Agno to power AI agents with web scraping, automation, and data insights.**

---

## What is Agno?

[Agno](https://docs.agno.com/) is an open-source framework for building intelligent AI agents. It provides a flexible architecture to create agents with custom tools, enabling seamless integration with external services like Apify for tasks such as web scraping, data extraction and automation.

:::note Agno documentation

Check out the [Agno documentation](https://docs.agno.com/introduction) for more details on building AI agents.

:::

## How to Use Apify with Agno

This guide shows how to integrate Apify Actors with Agno to empower your AI agents with real-time web data. We'll use the [RAG Web Browser](https://apify.com/apify/rag-web-browser) Actor to fetch web content and the [Google Places Crawler](https://apify.com/compass/crawler-google-places) Actor to extract location-based data. It is very easy to use with any other Actor by just passing the name of the Actor. See and choose from thousands of Actors in the [Apify Store](https://apify.com/store).

### Prerequisites

> - *Apify API token*: Obtain your token from the [Apify console](https://console.apify.com/account/integrations).  

Author: Should we require OpenAI API keys specifically, or should we also mention the possibility of supporting API keys from other providers?

> - *Apify API token*: Obtain your token from the [Apify console](https://console.apify.com/account/integrations).  

Reviewer 3: I think people will figure out they can use a different provider — I just wanted to make sure the code works as-is.

If you want to mention other providers, you’ll either confuse people or need to include an example.

> - *Apify API token*: Obtain your token from the [Apify console](https://console.apify.com/account/integrations).  

Reviewer: we can add note? something along the lines

:::tip Other providers

While our example uses OpenAI API specifically, you are free to use any provider

:::

or instead of separate admonition we can extend the tip in the prose, something along the lines of:

```
- *OpenAI API key*: Get your API key from the [OpenAI platform](https://platform.openai.com/account/api-keys). You can use any other provider as well
```

> - *Apify API token*: Obtain your token from the [Apify console](https://console.apify.com/account/integrations).  

Reviewer: from my POV great addition

> - *Apify API token*: Obtain your token from the [Apify console](https://console.apify.com/account/integrations).  
> - *Python environment*: Ensure Python is installed (version 3.8+ recommended).  
> - *Required packages*: Install the following dependencies in your terminal:

Reviewer: We use _ instead of * for emphasis

```suggestion
- _Apify API token_: Obtain your token from the [Apify console](https://console.apify.com/account/integrations).  
- _Python environment_: Ensure Python is installed (version 3.8+ recommended).  
- _Required packages_: Install the following dependencies in your terminal:
```

```bash
pip install agno apify-client
```

## Basic Integration Example

Start by setting up an Agno agent with Apify tools. This example uses the RAG Web Browser Actor to extract content from a specific URL.

```python
from agno.agent import Agent
from agno.tools.apify import ApifyTools

# Initialize the agent with Apify tools
agent = Agent(
    tools=[
        ApifyTools(
            actors=["apify/rag-web-browser"],
            apify_api_token="YOUR_APIFY_API_TOKEN"  # Replace YOUR_APIFY_API_TOKEN with your token 
        )
    ],
    show_tool_calls=True,
    markdown=True
)

# Fetch and display web content

> agent.print_response("Extract key details from https://docs.agno.com/introduction", markdown=True)

Author: Sure, or just let them choose the model they want. Doesn't have to be OpenAi.

    ],
    tools=[
        ApifyTools(
            actors=[
                "apify/rag-web-browser",           # For general web research
                "compass/crawler-google-places"    # For location-based data
            ],
            apify_api_token="YOUR_APIFY_API_TOKEN"
        )
    ],
    show_tool_calls=True,
    markdown=True
)

# Plan a trip to Tokyo
agent.print_response(
    """
    I'm traveling to Tokyo next month.
    1. Research the best time to visit and top attractions.
    2. Find a highly rated sushi restaurant near Shinjuku.
    Compile a travel guide with this information.
    """,
    markdown=True
)
```

This agent will fetch travel-related data and restaurant recommendations, providing a comprehensive travel guide:

1. Use the RAG Web Browser to research Tokyo travel details.
2. Use the Google Places Crawler to find a top sushi restaurant.
3. Combine the results into a comprehensive guide.

:::tip Apify Store

Browse the [Apify Store](https://apify.com/store) to find additional Actors for tasks like social media scraping, e-commerce data extraction, or news aggregation.

:::

> ### Available Apify Tools

Reviewer 3: ```suggestion
### Available Apify tools
```

Agno supports any Apify Actor via the ApifyTools class. You can specify a single Actor ID or a list of Actor IDs to register multiple tools for your agent at once.

> ## Configuration Options
> 
> | Parameter                    | Type                | Default | Description                                                        |
> | ---------------------------- | ------------------- | ------- | ------------------------------------------------------------------ |
> | `apify_api_token`            | `str`               | `None`  | Apify API token (or set via APIFY_API_TOKEN environment variable)  |
> | `actors`                     | `str` or `List[str]`| `None`  | Single Actor ID or list of Actor IDs to register                   |

Reviewer 3: ```suggestion
```

Reviewer 3: Do we really need this table?

I think the examples are enough. This table does not add much value IMO

Reviewer: IMO it gives more structure to information and makes it more scannable. But given that it's only two options I think this could also be remade to text.

    """
    I'm traveling to Tokyo next month.
    1. Research the best time to visit and top attractions.
    2. Find a highly rated sushi restaurant near Shinjuku.
    Compile a travel guide with this information.
    """,
    markdown=True
)
```

This agent will fetch travel-related data and restaurant recommendations, providing a comprehensive travel guide:

1. Use the RAG Web Browser to research Tokyo travel details.
2. Use the Google Places Crawler to find a top sushi restaurant.
3. Combine the results into a comprehensive guide.

:::tip Apify Store

Browse the [Apify Store](https://apify.com/store) to find additional Actors for tasks like social media scraping, e-commerce data extraction, or news aggregation.

:::

### Available Apify tools

Agno supports any Apify Actor via the ApifyTools class. You can specify a single Actor ID or a list of Actor IDs to register multiple tools for your agent at once.

## Configuration options

**`apify_api_token`** (string, default: `None`)
: Apify API token (or set via APIFY_API_TOKEN environment variable)

**`actors`** (string or List[string], default: `None`)
: Single Actor ID or list of Actor IDs to register

## Resources

> - [How to build an AI Agent](https://blog.apify.com/how-to-build-an-ai-agent/)

Reviewer 4: Hey folks, this blog does not feature Agno?

Can also include a direct link to this? https://docs.agno.com/tools/toolkits/others/apify#apify

Reviewer 3: Hey, yeah—it’s there because it demonstrates how to build and monetize an agent, which is relevant.

@ohlava Could you please include the link Priti shared?
